src/CoreRCON/Extensions.cs

```diff
@@ -134,18 +134,16 @@ public static async Task<T> TimeoutAfter<T>(this Task<T> task, TimeSpan? timeout
             {
                 var delayTask = Task.Delay(timeout.Value, cts.Token);
 
-                var resultTask = await Task.WhenAny(task, delayTask);
+                var resultTask = await Task.WhenAny(task, delayTask).ConfigureAwait(false);
                 if (resultTask == delayTask)
                 {
                     // Operation cancelled
                     throw new TimeoutException();
                 }
-                else
-                {
-                    cts.Cancel();
-                }
 
-                return await task;
```

Comment:
I don't understand this change,  we shouldn't have to cancel the cts if the delay task is completed.

Reply:
The code on the lefft and right are identical, the else is redundant because the if statement will throw